{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# should create a separate workbook 2.5. where I conduct the network analysis properly in Python --> also calculate weighted degree within communities"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#check whether I can still get the missing 1525 publications"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: folder=X:\\5_Research\\Paul\\dsi_origins\\\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from collections import Counter\n",
    "import pandas as pd \n",
    "import re\n",
    "import requests\n",
    "import json\n",
    "import time\n",
    "import glob\n",
    "import pprint\n",
    "import seaborn as sns\n",
    "import pylab as plt\n",
    "from matplotlib.ticker import MaxNLocator\n",
    "from matplotlib.colors import ListedColormap\n",
    "import pycountry \n",
    "import pycountry_convert as pc\n",
    "pp = pprint.PrettyPrinter()\n",
    "import numpy as np\n",
    "from tabulate import tabulate\n",
    "import geopandas as gpd\n",
    "from geopandas import GeoDataFrame\n",
    "import matplotlib.pyplot as plt\n",
    "from shapely.geometry import Point, Polygon\n",
    "import regex as re\n",
    "import plotly.graph_objects as go\n",
    "import os\n",
    "\n",
    "from string import punctuation\n",
    "from nltk.stem.snowball import SnowballStemmer\n",
    "from nltk.corpus import stopwords\n",
    "stemmer=SnowballStemmer(\"english\")\n",
    "import sklearn\n",
    "from sklearn.feature_extraction.text import CountVectorizer\n",
    "import math\n",
    "\n",
    "import networkx as nx\n",
    "#from fa2 import ForceAtlas2\n",
    "#import community as community_louvain  # For Louvain method\n",
    "\n",
    "from collections import defaultdict\n",
    "from itertools import combinations\n",
    "\n",
    "\n",
    "%env folder = X:\\5_Research\\Paul\\dsi_origins\\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = %env folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_communities_full = pd.read_csv(path+R'processed_data\\co_word_network_+2tfidf_backbone_90_communities_3.csv', index_col = 0)#.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(1, 8847), (0, 8420), (3, 7412), (2, 4)]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter(df_communities_full['modularity_class']).most_common()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#drop the one tiny community\n",
    "top_communities = [3,0,1]\n",
    "df_communities = df_communities_full[df_communities_full['modularity_class'].isin(top_communities)].rename(columns = {'Label':'word','modularity_class':'community','Weighted Degree':'weighted_degree'})#[~df_communities_full['community'].isin([23,22,13])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>timeset</th>\n",
       "      <th>weighted_degree</th>\n",
       "      <th>community</th>\n",
       "      <th>componentnumber</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>hiv1</th>\n",
       "      <td>hiv1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4235</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hepatitis</th>\n",
       "      <td>hepatitis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4219</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>distribution</th>\n",
       "      <td>distribution</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4080</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>influenza</th>\n",
       "      <td>influenza</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3991</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>effect</th>\n",
       "      <td>effect</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3595</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tigurinus</th>\n",
       "      <td>tigurinus</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>smallcolony</th>\n",
       "      <td>smallcolony</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asiapacific</th>\n",
       "      <td>asiapacific</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>empyema</th>\n",
       "      <td>empyema</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arenaviruses</th>\n",
       "      <td>arenaviruses</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24679 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      word  timeset  weighted_degree  community  \\\n",
       "Id                                                                \n",
       "hiv1                  hiv1      NaN             4235          3   \n",
       "hepatitis        hepatitis      NaN             4219          3   \n",
       "distribution  distribution      NaN             4080          0   \n",
       "influenza        influenza      NaN             3991          3   \n",
       "effect              effect      NaN             3595          1   \n",
       "...                    ...      ...              ...        ...   \n",
       "tigurinus        tigurinus      NaN                1          3   \n",
       "smallcolony    smallcolony      NaN                1          3   \n",
       "asiapacific    asiapacific      NaN                1          0   \n",
       "empyema            empyema      NaN                1          1   \n",
       "arenaviruses  arenaviruses      NaN                1          3   \n",
       "\n",
       "              componentnumber  \n",
       "Id                             \n",
       "hiv1                        0  \n",
       "hepatitis                   0  \n",
       "distribution                0  \n",
       "influenza                   0  \n",
       "effect                      0  \n",
       "...                       ...  \n",
       "tigurinus                   0  \n",
       "smallcolony                 0  \n",
       "asiapacific                 0  \n",
       "empyema                     0  \n",
       "arenaviruses                0  \n",
       "\n",
       "[24679 rows x 5 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_communities.sort_values('weighted_degree', ascending = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#reduce to top 10 words per community\n",
    "top_com_words = []\n",
    "for com in top_communities: \n",
    "    top_com_words.extend(list(df_communities[df_communities['community'] == com].sort_values('weighted_degree', ascending = False)['word'].values)[:20])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_communities_topwords = df_communities[df_communities['word'].isin(top_com_words)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>word</th>\n",
       "      <th>timeset</th>\n",
       "      <th>weighted_degree</th>\n",
       "      <th>community</th>\n",
       "      <th>componentnumber</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>hiv</th>\n",
       "      <td>hiv</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2808</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>highly</th>\n",
       "      <td>highly</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2622</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hiv1</th>\n",
       "      <td>hiv1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4235</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>response</th>\n",
       "      <td>response</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3260</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>multiple</th>\n",
       "      <td>multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2627</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>case</th>\n",
       "      <td>case</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2848</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>comparison</th>\n",
       "      <td>comparison</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2853</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>respiratory</th>\n",
       "      <td>respiratory</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2931</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>syndrome</th>\n",
       "      <td>syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2327</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hepatitis</th>\n",
       "      <td>hepatitis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4219</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>province</th>\n",
       "      <td>province</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2455</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>influenza</th>\n",
       "      <td>influenza</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3991</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>outbreak</th>\n",
       "      <td>outbreak</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3075</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sequencing</th>\n",
       "      <td>sequencing</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2997</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>reveals</th>\n",
       "      <td>reveals</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2710</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample</th>\n",
       "      <td>sample</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3028</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>viral</th>\n",
       "      <td>viral</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3514</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>united</th>\n",
       "      <td>united</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2411</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>state</th>\n",
       "      <td>state</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2945</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>vaccine</th>\n",
       "      <td>vaccine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2661</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    word  timeset  weighted_degree  community  componentnumber\n",
       "Id                                                                            \n",
       "hiv                  hiv      NaN             2808          3                0\n",
       "highly            highly      NaN             2622          3                0\n",
       "hiv1                hiv1      NaN             4235          3                0\n",
       "response        response      NaN             3260          3                0\n",
       "multiple        multiple      NaN             2627          3                0\n",
       "case                case      NaN             2848          3                0\n",
       "comparison    comparison      NaN             2853          3                0\n",
       "respiratory  respiratory      NaN             2931          3                0\n",
       "syndrome        syndrome      NaN             2327          3                0\n",
       "hepatitis      hepatitis      NaN             4219          3                0\n",
       "province        province      NaN             2455          3                0\n",
       "influenza      influenza      NaN             3991          3                0\n",
       "outbreak        outbreak      NaN             3075          3                0\n",
       "sequencing    sequencing      NaN             2997          3                0\n",
       "reveals          reveals      NaN             2710          3                0\n",
       "sample            sample      NaN             3028          3                0\n",
       "viral              viral      NaN             3514          3                0\n",
       "united            united      NaN             2411          3                0\n",
       "state              state      NaN             2945          3                0\n",
       "vaccine          vaccine      NaN             2661          3                0"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_communities_topwords[df_communities_topwords['community'] == 3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# start with pmid data\n",
    "with open(path+R'processed_data\\word_id_dict_+2_tfidf.json') as json_string:\n",
    "    word_id_dict = json.load(json_string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "word_list = list(df_communities['word'].dropna())\n",
    "\n",
    "pubs = []\n",
    "communities = []\n",
    "words = []\n",
    "\n",
    "for word in word_list:\n",
    "    community = list(df_communities['community'][df_communities['word'] == word])[0]\n",
    "    for pub_id in word_id_dict[word]:\n",
    "        pubs.append(pub_id)\n",
    "        communities.append(community)\n",
    "        words.append(word)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Series([], Name: community, dtype: int64)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_communities['community'][df_communities['word'] == word]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "97555"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(pubs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sem = pd.DataFrame(zip(pubs, communities, words)).reset_index().rename(columns = {1:'community', 0: 'id_select', 2:'word'})\n",
    "\n",
    "grouped = df_sem.groupby(['id_select','community']).size().reset_index(name='count')\n",
    "most_frequent = grouped.loc[grouped.groupby('id_select')['count'].idxmax()]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "words_sel_dict = {}\n",
    "for pub in set(df_sem['id_select']): \n",
    "    words_sel = list(df_sem['word'][df_sem['id_select'] == pub])\n",
    "    words_sel_str = ''\n",
    "    for word in words_sel: \n",
    "        if words_sel_str == '': \n",
    "            words_sel_str = word\n",
    "        else: \n",
    "            words_sel_str = words_sel_str+', ' +word\n",
    "    words_sel_dict[pub] = words_sel_str\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "most_frequent['words_sel'] = most_frequent['id_select'].map(words_sel_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "semantic_dict_3 = {\n",
    "    0: 'Evolutionary biology',\n",
    "    1: 'Molecular biology',\n",
    "    3: 'Public health'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#semantic_dict_6 = {\n",
    "#    3: 'Evolutionary biology',\n",
    "#    0: 'Molecular biology',\n",
    "#    1: 'Public health, respiratory diseases',\n",
    "#    2: 'Agriculture and diseases',\n",
    "#    5: 'Public health, viruses, vaccine'\n",
    "#    7: 'Public health, bacteria'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id_select</th>\n",
       "      <th>community</th>\n",
       "      <th>count</th>\n",
       "      <th>words_sel</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>61429</th>\n",
       "      <td>22345466.0</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>mucosal, envelope, repeat, correlate, acquisit...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49773</th>\n",
       "      <td>21044670.0</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>clone, protective, attenuate, subtypes, nonpat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147418</th>\n",
       "      <td>36129562.0</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>coinfection, dog, rate, detect, parvovirus, se...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166661</th>\n",
       "      <td>pmc4524230</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>express, interferon, alpha, highly, multiple, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166008</th>\n",
       "      <td>pmc4432077</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>method, antifungal, laser, matrixassisted, fli...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29559</th>\n",
       "      <td>17998550.0</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>h6, reverse, transcription, subtypes, rapid, r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39917</th>\n",
       "      <td>19605490.0</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>highly, hiv1, replication, recombinant, clinic...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93341</th>\n",
       "      <td>25976382.0</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>level, occur, loss, baseline, antiviral, corre...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22565</th>\n",
       "      <td>16806600.0</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>induce, response, immune, mouse, antigen, falc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72532</th>\n",
       "      <td>23578709.0</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>syndrome, sus, multisystemic, bocalike, postwe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106536</th>\n",
       "      <td>27626934.0</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>lethal, chimeric, allow, particle, animal, chi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98639</th>\n",
       "      <td>26560541.0</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>biochemical, method, laboratory, system, laser...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14399</th>\n",
       "      <td>15194768.0</td>\n",
       "      <td>3</td>\n",
       "      <td>17</td>\n",
       "      <td>identifies, tissue, envelope, biological, r5, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38346</th>\n",
       "      <td>19409946.0</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>exhibit, present, occur, expand, hiv1, amino, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85368</th>\n",
       "      <td>25019170.0</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>nucleotide, green, comparison, sybr, rapid, re...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4117</th>\n",
       "      <td>10400762.0</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>herpesvirus, major, across, multiple, locus, c...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109810</th>\n",
       "      <td>28243803.0</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>could, coinfection, form, norovirus, severe, e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88984</th>\n",
       "      <td>25445343.0</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>interferon, response, hcv, amino, ribavirin, i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35880</th>\n",
       "      <td>19075055.0</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>ccr5, reverse, envelope, nucleoside, combinati...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68866</th>\n",
       "      <td>23177564.0</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>frequent, persistent, low, occurrence, respira...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         id_select  community  count  \\\n",
       "61429   22345466.0          3     21   \n",
       "49773   21044670.0          3     21   \n",
       "147418  36129562.0          3     19   \n",
       "166661  pmc4524230          3     18   \n",
       "166008  pmc4432077          3     18   \n",
       "29559   17998550.0          3     17   \n",
       "39917   19605490.0          3     17   \n",
       "93341   25976382.0          3     17   \n",
       "22565   16806600.0          3     17   \n",
       "72532   23578709.0          3     17   \n",
       "106536  27626934.0          3     17   \n",
       "98639   26560541.0          3     17   \n",
       "14399   15194768.0          3     17   \n",
       "38346   19409946.0          3     16   \n",
       "85368   25019170.0          3     16   \n",
       "4117    10400762.0          3     16   \n",
       "109810  28243803.0          3     16   \n",
       "88984   25445343.0          3     16   \n",
       "35880   19075055.0          3     16   \n",
       "68866   23177564.0          3     16   \n",
       "\n",
       "                                                words_sel  \n",
       "61429   mucosal, envelope, repeat, correlate, acquisit...  \n",
       "49773   clone, protective, attenuate, subtypes, nonpat...  \n",
       "147418  coinfection, dog, rate, detect, parvovirus, se...  \n",
       "166661  express, interferon, alpha, highly, multiple, ...  \n",
       "166008  method, antifungal, laser, matrixassisted, fli...  \n",
       "29559   h6, reverse, transcription, subtypes, rapid, r...  \n",
       "39917   highly, hiv1, replication, recombinant, clinic...  \n",
       "93341   level, occur, loss, baseline, antiviral, corre...  \n",
       "22565   induce, response, immune, mouse, antigen, falc...  \n",
       "72532   syndrome, sus, multisystemic, bocalike, postwe...  \n",
       "106536  lethal, chimeric, allow, particle, animal, chi...  \n",
       "98639   biochemical, method, laboratory, system, laser...  \n",
       "14399   identifies, tissue, envelope, biological, r5, ...  \n",
       "38346   exhibit, present, occur, expand, hiv1, amino, ...  \n",
       "85368   nucleotide, green, comparison, sybr, rapid, re...  \n",
       "4117    herpesvirus, major, across, multiple, locus, c...  \n",
       "109810  could, coinfection, form, norovirus, severe, e...  \n",
       "88984   interferon, response, hcv, amino, ribavirin, i...  \n",
       "35880   ccr5, reverse, envelope, nucleoside, combinati...  \n",
       "68866   frequent, persistent, low, occurrence, respira...  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "most_frequent[most_frequent['community'] ==3].sort_values('count', ascending = False)[:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'class': 456,\n",
       "         'pneumoniae': 390,\n",
       "         'element': 349,\n",
       "         'antibiotic': 351,\n",
       "         'chromosome': 261,\n",
       "         'resistant': 287,\n",
       "         'carry': 278,\n",
       "         'streptococcus': 366,\n",
       "         'bacteriophage': 407,\n",
       "         'escherichia': 853,\n",
       "         'hospital': 342,\n",
       "         'mycobacterium': 331,\n",
       "         'acinetobacter': 367,\n",
       "         'klebsiella': 363,\n",
       "         'phage': 364,\n",
       "         'plasmid': 693,\n",
       "         'salmonella': 309,\n",
       "         'subsp': 331,\n",
       "         'pseudomonas': 880,\n",
       "         'vibrio': 417})"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Counter(df_sem['word'][df_sem['community'] == 3])#.sort_values('count', ascending = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-25-f4e683916dd1>:2: DtypeWarning: Columns (11,12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df_scholarly = pd.read_csv(path+R'processed_data\\scholarly_aff_country.csv', index_col = 0)\n"
     ]
    }
   ],
   "source": [
    "#get the original titles with the id selects\n",
    "df_scholarly = pd.read_csv(path+R'processed_data\\scholarly_aff_country.csv', index_col = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_id_title = df_scholarly[['id_select', 'title']].drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_id_title.to_csv(path+R'processed_data\\id_title_dict.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id_select</th>\n",
       "      <th>community</th>\n",
       "      <th>count</th>\n",
       "      <th>words_sel</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002/ece3.2809</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>change, environment, photobiont, lichen, accli...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1002/jpln.201100382</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>involve, fertilization, potentially</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10.1002/jsfa.8281</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>ensifer, symbiovar, nodulates, vigna, unguiculata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10.1002/mbo3.47</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>cold, gulf, metazoan, seep, site, chemosynthet...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10.1002/mrd.22210</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>giant, interact, able, oocyte, erk, starfish</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187368</th>\n",
       "      <td>pmc93297</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>iron, across, cyanobacterial, deficiency, probe</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187370</th>\n",
       "      <td>pmc93301</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>active, nitriteoxidizing, situ, treatment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187373</th>\n",
       "      <td>pmc96788</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>express, major, borrelia, blood, tissue, mouse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187375</th>\n",
       "      <td>pmc97813</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>diverse, escherichia, iv, enteropathogenic, pilin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187376</th>\n",
       "      <td>pmc98260</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>major, tick, anaplasma, antigenic, marginale, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>97555 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     id_select  community  count  \\\n",
       "0            10.1002/ece3.2809          0      6   \n",
       "3       10.1002/jpln.201100382          1      2   \n",
       "5            10.1002/jsfa.8281          1      5   \n",
       "7              10.1002/mbo3.47          1      4   \n",
       "9            10.1002/mrd.22210          0      2   \n",
       "...                        ...        ...    ...   \n",
       "187368                pmc93297          1      3   \n",
       "187370                pmc93301          1      3   \n",
       "187373                pmc96788          3      7   \n",
       "187375                pmc97813          3      4   \n",
       "187376                pmc98260          0      3   \n",
       "\n",
       "                                                words_sel  \n",
       "0       change, environment, photobiont, lichen, accli...  \n",
       "3                     involve, fertilization, potentially  \n",
       "5       ensifer, symbiovar, nodulates, vigna, unguiculata  \n",
       "7       cold, gulf, metazoan, seep, site, chemosynthet...  \n",
       "9            giant, interact, able, oocyte, erk, starfish  \n",
       "...                                                   ...  \n",
       "187368    iron, across, cyanobacterial, deficiency, probe  \n",
       "187370          active, nitriteoxidizing, situ, treatment  \n",
       "187373  express, major, borrelia, blood, tissue, mouse...  \n",
       "187375  diverse, escherichia, iv, enteropathogenic, pilin  \n",
       "187376  major, tick, anaplasma, antigenic, marginale, ...  \n",
       "\n",
       "[97555 rows x 4 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "most_frequent"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id_select</th>\n",
       "      <th>title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>32373149.0</td>\n",
       "      <td>Evolution and Expression of Reproductive Trans...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>19763271.0</td>\n",
       "      <td>Limited neutralizing antibody specificities dr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>29324085.0</td>\n",
       "      <td>Hyperparasitism of an Avian Ectoparasitic Hipp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>30727098.0</td>\n",
       "      <td>First Report of Amphobotrys ricini Causing Gra...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>10406114.0</td>\n",
       "      <td>The divergence-homogenization duality in the e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>447178</th>\n",
       "      <td>pmc1590031</td>\n",
       "      <td>Development and validation of a bovine macroph...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>447188</th>\n",
       "      <td>pmc4766875</td>\n",
       "      <td>Borrelia miyamotoi and Candidatus Neoehrlichia...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>447204</th>\n",
       "      <td>pmc1352205</td>\n",
       "      <td>Real-time PCR for the detection and quantifica...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>447207</th>\n",
       "      <td>pmc6296573</td>\n",
       "      <td>Purification and characterization of lipase by...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>447211</th>\n",
       "      <td>pmc3218519</td>\n",
       "      <td>Comparative Study of different msDNA (multicop...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99179 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         id_select                                              title\n",
       "0       32373149.0  Evolution and Expression of Reproductive Trans...\n",
       "4       19763271.0  Limited neutralizing antibody specificities dr...\n",
       "14      29324085.0  Hyperparasitism of an Avian Ectoparasitic Hipp...\n",
       "18      30727098.0  First Report of Amphobotrys ricini Causing Gra...\n",
       "23      10406114.0  The divergence-homogenization duality in the e...\n",
       "...            ...                                                ...\n",
       "447178  pmc1590031  Development and validation of a bovine macroph...\n",
       "447188  pmc4766875  Borrelia miyamotoi and Candidatus Neoehrlichia...\n",
       "447204  pmc1352205  Real-time PCR for the detection and quantifica...\n",
       "447207  pmc6296573  Purification and characterization of lipase by...\n",
       "447211  pmc3218519  Comparative Study of different msDNA (multicop...\n",
       "\n",
       "[99179 rows x 2 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_id_title"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "#merge original title with id_select\n",
    "community_analysis = pd.merge(most_frequent,df_id_title, how = 'left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id_select</th>\n",
       "      <th>community</th>\n",
       "      <th>count</th>\n",
       "      <th>words_sel</th>\n",
       "      <th>title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002/ece3.2809</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>change, environment, photobiont, lichen, accli...</td>\n",
       "      <td>Lichen acclimation to changing environments: P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002/jpln.201100382</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>involve, fertilization, potentially</td>\n",
       "      <td>Characterization of new soil thermophilic bact...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002/jsfa.8281</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>ensifer, symbiovar, nodulates, vigna, unguiculata</td>\n",
       "      <td>A novel symbiovar (aegeanense) of the genus En...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1002/mbo3.47</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>cold, gulf, metazoan, seep, site, chemosynthet...</td>\n",
       "      <td>Diversity of symbioses between chemosynthetic ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1002/mrd.22210</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>giant, interact, able, oocyte, erk, starfish</td>\n",
       "      <td>SGEBP, a giant protein from starfish oocytes a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97550</th>\n",
       "      <td>pmc93297</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>iron, across, cyanobacterial, deficiency, probe</td>\n",
       "      <td>Detection of the isiA Gene across Cyanobacteri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97551</th>\n",
       "      <td>pmc93301</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>active, nitriteoxidizing, situ, treatment</td>\n",
       "      <td>In Situ Characterization of Nitrospira-Like Ni...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97552</th>\n",
       "      <td>pmc96788</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>express, major, borrelia, blood, tissue, mouse...</td>\n",
       "      <td>Characterization of VspB of Borrelia turicatae...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97553</th>\n",
       "      <td>pmc97813</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>diverse, escherichia, iv, enteropathogenic, pilin</td>\n",
       "      <td>Molecular variation among type IV pilin (bfpA)...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97554</th>\n",
       "      <td>pmc98260</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>major, tick, anaplasma, antigenic, marginale, ...</td>\n",
       "      <td>Antigenic Variation of Anaplasma Marginale: Ma...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>97555 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    id_select  community  count  \\\n",
       "0           10.1002/ece3.2809          0      6   \n",
       "1      10.1002/jpln.201100382          1      2   \n",
       "2           10.1002/jsfa.8281          1      5   \n",
       "3             10.1002/mbo3.47          1      4   \n",
       "4           10.1002/mrd.22210          0      2   \n",
       "...                       ...        ...    ...   \n",
       "97550                pmc93297          1      3   \n",
       "97551                pmc93301          1      3   \n",
       "97552                pmc96788          3      7   \n",
       "97553                pmc97813          3      4   \n",
       "97554                pmc98260          0      3   \n",
       "\n",
       "                                               words_sel  \\\n",
       "0      change, environment, photobiont, lichen, accli...   \n",
       "1                    involve, fertilization, potentially   \n",
       "2      ensifer, symbiovar, nodulates, vigna, unguiculata   \n",
       "3      cold, gulf, metazoan, seep, site, chemosynthet...   \n",
       "4           giant, interact, able, oocyte, erk, starfish   \n",
       "...                                                  ...   \n",
       "97550    iron, across, cyanobacterial, deficiency, probe   \n",
       "97551          active, nitriteoxidizing, situ, treatment   \n",
       "97552  express, major, borrelia, blood, tissue, mouse...   \n",
       "97553  diverse, escherichia, iv, enteropathogenic, pilin   \n",
       "97554  major, tick, anaplasma, antigenic, marginale, ...   \n",
       "\n",
       "                                                   title  \n",
       "0      Lichen acclimation to changing environments: P...  \n",
       "1      Characterization of new soil thermophilic bact...  \n",
       "2      A novel symbiovar (aegeanense) of the genus En...  \n",
       "3      Diversity of symbioses between chemosynthetic ...  \n",
       "4      SGEBP, a giant protein from starfish oocytes a...  \n",
       "...                                                  ...  \n",
       "97550  Detection of the isiA Gene across Cyanobacteri...  \n",
       "97551  In Situ Characterization of Nitrospira-Like Ni...  \n",
       "97552  Characterization of VspB of Borrelia turicatae...  \n",
       "97553  Molecular variation among type IV pilin (bfpA)...  \n",
       "97554  Antigenic Variation of Anaplasma Marginale: Ma...  \n",
       "\n",
       "[97555 rows x 5 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "community_analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Molecular phylogeny and DNA barcoding of tropical eastern Pacific shallow-water gorgonian octocorals'"
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "community_analysis['title'][3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "community_analysis.to_csv(path+R'processed_data\\community_analysis_3.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# add the communities to the regression dataframe\n",
    "df_extract = pd.read_csv(path+R'processed_data\\extractivism_test.csv', index_col = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_extract_com = pd.merge(df_extract, community_analysis[['id_select','community']], how = 'left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_extract_com.to_csv(path+R'processed_data\\extractivism_com3.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id_select</th>\n",
       "      <th>extractivism_oecd_1</th>\n",
       "      <th>extractivism_brics_1</th>\n",
       "      <th>extracts_oecd_2_test</th>\n",
       "      <th>extractivism_brics_2</th>\n",
       "      <th>cooperation_oecd_1</th>\n",
       "      <th>cooperation_brics_2</th>\n",
       "      <th>community</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002/ece3.2809</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002/jpln.201100382</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002/jsfa.8281</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1002/mbo3.47</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1002/mrd.22210</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95403</th>\n",
       "      <td>pmc93297</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95404</th>\n",
       "      <td>pmc93301</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95405</th>\n",
       "      <td>pmc96788</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95406</th>\n",
       "      <td>pmc97813</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95407</th>\n",
       "      <td>pmc98260</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>93883 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    id_select  extractivism_oecd_1  extractivism_brics_1  \\\n",
       "0           10.1002/ece3.2809                  1.0                   0.0   \n",
       "1      10.1002/jpln.201100382                  0.0                   0.0   \n",
       "2           10.1002/jsfa.8281                  0.0                   0.0   \n",
       "3             10.1002/mbo3.47                  1.0                   0.0   \n",
       "4           10.1002/mrd.22210                  0.0                   0.0   \n",
       "...                       ...                  ...                   ...   \n",
       "95403                pmc93297                  1.0                   0.0   \n",
       "95404                pmc93301                  0.0                   0.0   \n",
       "95405                pmc96788                  0.0                   0.0   \n",
       "95406                pmc97813                  1.0                   0.0   \n",
       "95407                pmc98260                  0.0                   0.0   \n",
       "\n",
       "       extracts_oecd_2_test  extractivism_brics_2  cooperation_oecd_1  \\\n",
       "0                         1                     0                   0   \n",
       "1                         0                     0                   0   \n",
       "2                         0                     0                   0   \n",
       "3                         1                     0                   0   \n",
       "4                         0                     0                   0   \n",
       "...                     ...                   ...                 ...   \n",
       "95403                     1                     0                   0   \n",
       "95404                     0                     0                   0   \n",
       "95405                     0                     0                   0   \n",
       "95406                     1                     0                   0   \n",
       "95407                     0                     0                   0   \n",
       "\n",
       "       cooperation_brics_2  community  \n",
       "0                        0        0.0  \n",
       "1                        0        1.0  \n",
       "2                        0        1.0  \n",
       "3                        0        1.0  \n",
       "4                        0        0.0  \n",
       "...                    ...        ...  \n",
       "95403                    0        1.0  \n",
       "95404                    0        1.0  \n",
       "95405                    0        3.0  \n",
       "95406                    0        3.0  \n",
       "95407                    0        0.0  \n",
       "\n",
       "[93883 rows x 8 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_extract_com[df_extract_com['community'].notna()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### trash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "df_grtype_scholarly_filt = df_grtype_scholarly[['pub_id_sel','GROUP_dsi','GROUP_lit']]\n",
    "df_grtype_scholarly_semantics = pd.merge(df_grtype_scholarly, most_frequent[['pub_id_sel','sem_community']], on = 'pub_id_sel', how = 'inner')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "semantic_dict = {\n",
    "    10: 'Public health',\n",
    "    4: 'Evolutionary biology',\n",
    "    1: 'Environmental microbiology',\n",
    "    2: 'Molecular biology',\n",
    "    14: 'Species identification',\n",
    "    6: 'Agriculture and diseases',\n",
    "    -2: 'No semantic leaning'\n",
    "}\n",
    "\n",
    "df_grtype_scholarly_semantics['semantic_name'] = df_grtype_scholarly_semantics['sem_community'].map(semantic_dict)\n",
    "\n",
    "# semantics\n",
    "# Get unique types of 'semantics' after dropping NaNs\n",
    "semantics = sorted(set(df_grtype_scholarly_semantics['semantic_name'].dropna()))\n",
    "\n",
    "# Number of subplots\n",
    "n = len(semantics)\n",
    "\n",
    "# Calculate rows and columns for the grid\n",
    "n_cols = 2\n",
    "n_rows = (n + 1) // n_cols  # Ensures correct number of rows for odd n\n",
    "\n",
    "# Create subplots with 2 rows and multiple columns\n",
    "fig, axes = plt.subplots(n_rows, n_cols, figsize=(10, 10), sharey=False)\n",
    "\n",
    "# Flatten axes array for easier iteration\n",
    "axes = axes.flatten()\n",
    "\n",
    "# Set the theme for the plot\n",
    "sns.set_theme(style=\"darkgrid\")\n",
    "\n",
    "# Define the palette for the different groups\n",
    "palette = {\"G77\": \"tab:purple\", \"OECD\": \"tab:blue\", \"BRICS\": \"tab:red\"}\n",
    "\n",
    "for i, semantic in enumerate(semantics):\n",
    "    vis_df_subgroup = df_grtype_scholarly_semantics[df_grtype_scholarly_semantics['semantic_name'] == semantic]\n",
    "    vis_df_subgroup_count1pg = vis_df_subgroup[['pub_id_sel','GROUP_dsi','GROUP_lit']].drop_duplicates().rename(columns = {'GROUP_dsi':'GROUP_dsi'}).reset_index(drop=True).sort_values(['GROUP_lit'], ascending = False)\n",
    "\n",
    "    sns.countplot(\n",
    "        data=vis_df_subgroup_count1pg,\n",
    "        x=\"GROUP_dsi\", \n",
    "        hue=\"GROUP_lit\",\n",
    "        order = ['OECD','BRICS','G77'],\n",
    "        palette=palette, \n",
    "        ax=axes[i]\n",
    "    )\n",
    "\n",
    "    # Set labels and title\n",
    "    axes[i].set_title(semantic, fontsize=14)\n",
    "    axes[i].set_xlabel('GROUP_dsi (providers)', fontsize=12)\n",
    "    axes[i].set_ylabel('Count', fontsize=12)\n",
    "    axes[i].tick_params(axis='x', labelsize=10)\n",
    "    axes[i].tick_params(axis='y', labelsize=10)\n",
    "\n",
    "# Hide any unused subplots\n",
    "for j in range(i + 1, len(axes)):\n",
    "    axes[j].set_visible(False)\n",
    "\n",
    "plt.tight_layout()\n",
    "\n",
    "# Save the combined figure\n",
    "plt.savefig(path + R'plots\\coo_cor_combined_semantic.svg', format=\"svg\")\n",
    "plt.savefig(path + R'plots\\coo_cor_combined_semantic.jpg', format=\"jpg\")\n",
    "\n",
    "plt.show()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:base] *",
   "language": "python",
   "name": "conda-base-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
